### DISCUSSION TITLE
Neovim will remove tar from their releases soon

### DISCUSSION THREAD

**Discussion by dundargoc** [MordechaiHadad/bob#88] [General]
Hey. People told me bob uses the tars in the release page and wanted to give a heads up that they might get removed soon. Just wanted to give a heads up.

**Comment by MordechaiHadad** [comment]
@dundargoc thank you I wasn't aware, are they going to use zip instead?

**Reply by dundargoc** [reply]
I think the tar and deb will be removed and only the appimage will be provided. I'm only talking about Linux releases.

**Reply by MordechaiHadad** [reply]
Bruh why would they do that, at least remove the deb, smh

**Reply by glacambre** [reply]
@MordechaiHadad In case you don't know, the appimage is an archive itself, `./nvim.appimage --appimage-extract` is equivalent to untarring the tar.

**Reply by MordechaiHadad** [reply]
@glacambre Oh? so I can basically extract the content of the archive like zip, tar, rar etc?

**Reply by glacambre** [reply]
Yep! Although I think most of the time you won't need to extract it, the appimage is a self-contained neovim binary that can be run on its own, without any installation steps. Just make it executable and drop it in your $PATH and you're done :)

**Reply by MordechaiHadad** [reply]
Oh I see, thank you.
I hope I will be able to update it in time in case it happens.

**Reply by dundargoc** [reply]
How much time do you need?

**Reply by dundargoc** [reply]
I think I can convince the Neovim team to hold off from removing it if you need some time.

**Reply by MordechaiHadad** [reply]
@dundargoc I will just change it whenever it happens I want to take a break from bob for now.
Thanks anyways

**Reply by MordechaiHadad** [reply]
If it's possible by you just post on this discussion whenever this changes occurs so I can be aware of it asap

**Reply by dundargoc** [reply]
Uh, sure. If I don't forget I will.

**Comment by dundargoc** [comment]
@MordechaiHadad preliminary schedule is that tar.gz will be removed by 0.9, although I'd take that with pinch of salt (meaning, start hustling now if you can).

https://github.com/neovim/neovim/issues/22684

**Reply by MordechaiHadad** [reply]
@dundargoc thank you for the heads up!

**Reply by MordechaiHadad** [reply]
Btw I didn't ask, but why did the neovim team decided to keep the appimage instead of the .tar.gz file, also can an appimage that was compiled in an x86 environment be ran inside of an arm machine? I have never used appimages before.

EDIT: I asked chatgpt and it doesn't seem like it

**Reply by dundargoc** [reply]
Appimages are more versatile, they can be used on its own but also extracted like with tar.gz. I will investigate how and document properly in the issue/PR.

x86 appimages won't work on arm.

**Reply by MordechaiHadad** [reply]
@dundargoc thank you!

**Reply by clason** [reply]
To be clear, `.tar.gz` will be removed _after_ 0.9; it's only `.deb` that needs to die as soon as possible (which is the 0.9 release). Do let us know if there are issues with the appimage that prevent you from using it.

**Reply by clason** [reply]
And the [Neovim releases page](https://github.com/neovim/neovim/releases/tag/nightly) has a snippet for using appimages, both directly and with extraction -- again, let us know if that is not sufficiently clear!

(As @dundargoc mentioned, appimages are much more robust against people moving them around -- they keep forgetting that the runtime needs to be where the binary expects it.)

**Reply by MordechaiHadad** [reply]
@clason I will thank you! from my current estimation I don't see why it would be too much of an issue, but I will keep you guys posted.

**Comment by MordechaiHadad** [comment]
Seems like appimages require FUSE: https://github.com/AppImage/AppImageKit/wiki/FUSE
So I will make bob extract it

**Comment by MordechaiHadad** [comment]
bob is now using appimages for linux

**Reply by clason** [reply]
But note that we have decided to still package the appimage build as `tar.gz`, so that will still work for the foreseeable future.

**Reply by MordechaiHadad** [reply]
lmao what caused this change of mind?

**Reply by clason** [reply]
Loud complaints, and we didn't want to break bob (which looked like it would keep relying on it for a longer time).

(The sticking point was that these used to be separate builds with subtly different properties; now they are the same build, just packaged differently. That seemed like a fair compromise.)

**Reply by MordechaiHadad** [reply]
I see, fair enough lol seems like people like using tarballs.

**Reply by clason** [reply]
They really do 😆  (For not always comprehensible reasons.)

**Reply by MordechaiHadad** [reply]
we should normalize only using zips(for no comprehensible reason)

btw any plans on providing arm builds in the future?

**Reply by clason** [reply]
> we should normalize only using zips(for no comprehensible reason)

There are very strong reasons why we provide appimages (for one, having the correct runtime bundled and in the right place; if I had a cookie for every time someone unzipped the tar, moved the `nvim` binary, and then complained to us that things don't work...)

> btw any plans on providing arm builds in the future?

None, as we don't have any ARM runners and no support for cross-compilation. Someone would have to set up a docker environment and make a PR.

**Reply by MordechaiHadad** [reply]
> None, as we don't have any ARM runners and no support for cross-compilation. Someone would have to set up a docker environment and make a PR.
that's true but C has arm compilers no? this is how I made bob have arm support while not having arm runners

**Reply by clason** [reply]
Cross-compilation is of course possible _in general_; it's just not possible for Neovim (our -- already fairly complex -- build framework doesn't support it). This means the only way to build Neovim _for_ ARM is to build _on_ ARM -- hence the requirement for ARM runners or containers.

**Reply by MordechaiHadad** [reply]
Oh... shame.
Aight once arm runners are out i'll try to stay updated and add the neovim arm releases to bob

**Reply by clason** [reply]
I wouldn't hold my breath on that one; I just mentioned it for the sake of completeness. ARM releases will happen if (and only if) someone makes a PR with a docker setup (which we already use for building on Ubuntu 18.04 for those stuck on outdated distros).

**Reply by MordechaiHadad** [reply]
I wouldn't! unless I swim.
thank you clason 🤝
